source/common/upstream/upstream_impl.cc

```diff
@@ -1342,8 +1343,12 @@ Network::UpstreamTransportSocketFactoryPtr createTransportSocketFactory(
 }
 
 void ClusterInfoImpl::createNetworkFilterChain(Network::Connection& connection) const {
-  for (const auto& factory : filter_factories_) {
-    factory(connection);
+  for (const auto& filter_config_provider : filter_factories_) {
+    auto config = filter_config_provider->config();
+    if (config.has_value()) {
+      auto factory = config.value();
```

Comment:
Can I suggest to spell out the type? I want to make sure a possible implicit copy is not expensive.

Reply:
done